examples/methods/plot_subspace.py

```diff
@@ -0,0 +1,264 @@
+# ------------------------------------------
+#
+# We generate a simple 2D dataset with subspace shift.
+
+RANDOM_SEED = 42
+
+X, y, sample_domain = make_shifted_datasets(
+    n_samples_source=20,
+    n_samples_target=20,
+    noise=0.1,
+    random_state=RANDOM_SEED,
+    shift="subspace",
+)
+
+Xs, Xt, ys, yt = source_target_split(X, y, sample_domain=sample_domain)
+
+# %%
+# Plot of the dataset:
+# ------------------------------------------
+
+x_min, x_max = -2.4, 2.4
+y_min, y_max = -2.4, 2.4
+
+
+figsize = (8, 4)
+figure, axes = plt.subplots(1, 2, figsize=figsize)
+
+cm = plt.cm.RdBu
+colormap = ListedColormap(["#FFA056", "#6C4C7C"])
+ax = axes[0]
+ax.set_title("Source data")
+# Plot the source points:
+ax.scatter(Xs[:, 0], Xs[:, 1], c=ys, cmap=colormap, alpha=0.7, s=[15])
+
+ax.set_xticks(()), ax.set_yticks(())
+ax.set_xlim(x_min, x_max), ax.set_ylim(y_min, y_max)
+
+ax = axes[1]
+
+ax.set_title("Target data")
+# Plot the target points:
+ax.scatter(Xt[:, 0], Xt[:, 1], c=ys, cmap=colormap, alpha=0.1, s=[15])
+ax.scatter(Xt[:, 0], Xt[:, 1], c=yt, cmap=colormap, alpha=0.7, s=[15])
+figure.suptitle("Plot of the dataset", fontsize=16, y=1)
+ax.set_xticks(()), ax.set_yticks(())
+ax.set_xlim(x_min, x_max), ax.set_ylim(y_min, y_max)
+
+# %%
+#     Illustration of the problem with no domain adaptation
+# ------------------------------------------
+#
+# When not using domain adaptation, the classifier won't train on
+# data that is distributed as the target sample domain, it will thus
+# not be performing optimaly.
+
+# We create a dict to store scores:
+scores_dict = {}
+
+
+def create_plots(
+    clf,
+    name="Without DA",
+    suptitle=None,
+):
+    size = np.array([16] * Xs.shape[0])
+
+    if suptitle is None:
+        suptitle = f"Illustration of the {name} method"
+    figure, axes = plt.subplots(1, 3, figsize=figsize)
+    ax = axes[1]
+    score = clf.score(Xt, yt)
+    DecisionBoundaryDisplay.from_estimator(
+        clf,
+        Xs,
+        cmap=colormap,
+        alpha=0.1,
+        ax=ax,
+        eps=0.5,
+        response_method="predict",
+    )
+
+    # Plot the target points:
+    ax.scatter(
+        Xt[:, 0],
+        Xt[:, 1],
+        c=yt,
+        cmap=colormap,
+        alpha=0.7,
+        s=[15],
+    )
+
+    ax.set_xticks(()), ax.set_yticks(())
+    ax.set_xlim(x_min, x_max), ax.set_ylim(y_min, y_max)
+    ax.set_title("Accuracy on target", fontsize=12)
+    ax.text(
+        x_max - 0.3,
+        y_min + 0.3,
+        ("%.2f" % score).lstrip("0"),
+        size=15,
+        horizontalalignment="right",
+    )
+    scores_dict[name] = score
+
+    ax = axes[0]
+
+    # Plot the source points:
+    ax.scatter(Xs[:, 0], Xs[:, 1], c=ys, cmap=colormap, alpha=0.7, s=size)
+
+    ax.set_xticks(()), ax.set_yticks(())
+    ax.set_xlim(x_min, x_max), ax.set_ylim(y_min, y_max)
+    ax.set_title("Training with rewegihted data", fontsize=12)
+    if name != "Without DA":
+        ax = axes[2]
```

Comment:
pMaybe it could be nice to see how to get the subspace, and not hide it in the function